### PR TITLE
Rails 3.2.13 upgrade

### PR DESCRIPTION
**Pull request by isratrade**
You must do "rake assets:precompile" if deploying in production mode

### CODE REVIEW

File: app/models/host/base.rb [hunk 1]
```diff
@@ -14,7 +14,9 @@ class Base < ActiveRecord::Base
     validates_uniqueness_of :name
     validate :is_name_downcased?
 
-    include Hostext::Search
```
Comment (GregSutcliffe) [inline app/models/host/base.rb:17]: Are we going to put this back in Base later? I'd like to be able to use search for host plugins too :)

**Comment by isratrade** [comment]
@GregSutcliffe I think we need to wait for @abenari to update the scoped_search gem before we move "include Hostext::Search" to Host::Base.  It broke the search when it was there.

**Comment by ares** [comment]
One of commits removed toggle_div helper however it's still being used in https://github.com/theforeman/foreman/blob/develop/app/views/common/_notice.erb#L20

However I'm not sure whether _notice partial is being used at all.

**Comment by GregSutcliffe** [comment]
@isratrade thats fine, i'm happy to merge as is to get rails32 moving forward - there's no code using search in the discovery plugin yet anyway.

**Comment by isratrade** [comment]
thanks @ares.  Notices currently aren't being used.  I assume that this code can be removed.  What do you think @ohadlevy ?

**Comment by xsuchy** [comment]
Tested, still getting:
```Started GET "/foreman/users/login" for 127.0.0.1 at 2013-03-27 08:46:40 -0400
Processing by UsersController#login as HTML
  Rendered users/login.html.erb within layouts/application (68.4ms)
Operation FAILED: foreman_large.png isn't precompiled
  Rendered common/500.html.erb within layouts/application (3.4ms)
Completed 500 Internal Server Error in 149ms

ActionView::Template::Error (application.css isn't precompiled):
    2: <html>
    3:   <head>
    4:     <title> <%= h(yield(:title) || "Foreman") %></title>
    5:     <%= stylesheet_link_tag 'application' %>
    6:     <%= javascript_include_tag 'application' %>
    7:     <%= csrf_meta_tags %>
    8:     <%= javascript_tag "var AUTH_TOKEN = #{form_authenticity_token.inspect};" if protect_against_forgery? %>
  app/views/layouts/application.html.erb:5:in `_app_views_layouts_application_html_erb___3463191502799506874_31749080'
  app/controllers/application_controller.rb:291:in`generic_exception'

```
```

**Comment by isratrade** [comment]
@xsuchy, in production mode, you need to run "rake assets:precompile".  This needs to be documented when merged.

**Comment by xsuchy** [comment]
```
[root@intel-piketon-02 foreman]# rake assets:precompile
(in /usr/share/foreman)
rake aborted!
undefined method `assets' for #<Rails::Application::Configuration:0x7fc61d6d4968>
/usr/share/foreman/Rakefile:1
(See full trace by running task with --trace)
```

I'm using foreman installed from .rpm package

**Comment by isratrade** [comment]
@GregSutcliffe @skottler , any ideas why @xsuchy can't run rake assets:precompile if using RPM

@xsuchy, try to run "bundle update rails"

**Comment by xsuchy** [comment]
Hmm that can be because we do not have packaged gems from "bundler.d/assets.rb". Hmm we will need to package it and make it available in build time and edit spec file to call"
  rake assets:precompile
in %build phase

**Comment by skottler** [comment]
I'll add that to the RPM's today. Right now we put individual files from `bundler.d` into their own subpackages so I will create an additional one for assets and then add a require just to keep with the existing pattern. How does that sound @xsuchy?

**Comment by xsuchy** [comment]
@skottler good

**Comment by skottler** [comment]
Cool, I'll work on that ASAP.

**Comment by GregSutcliffe** [comment]
i'm guessing i'll need to do the same for debian. @skottler If it's required for foreman to function at all, why bother making it a separate package? I added i18n.rb to the core package, I see no reason not to to the same with assets.rb.

**Comment by skottler** [comment]
@GregSutcliffe because then I've introduced a new pattern into the RPM's and there are two separate ways the files get managed. I would rather have them individually packaged for consistency's sake.

**Comment by skottler** [comment]
I opened a pull request with the fix for the RPM's at theforeman/foreman-rpms#53.

**Comment by ohadlevy** [comment]
Sam, dont forget we need to add the precompile task to the package build?

**Comment by GregSutcliffe** [comment]
@abenari @isratrade There's still at least one JS error in the Host Edit page. Easy to reproduce - edit a host and change the hostgroup or environment. You'll get a POST like this:

```
Started POST "/hosts/hostgroup_or_environment_selected" for 127.0.0.1 at Thu Mar 28 11:22:10 +0000 2013
Processing by HostsController#hostgroup_or_environment_selected as */*
  Parameters: {"hostgroup_id"=>"undefined", "environment_id"=>"undefined", "host_id"=>"1"}
```

Somehow the form is sending "undefined" instead of the hostgroup or environment ID. I'd fix it but I know nothing about JS :)

**Comment by isratrade** [comment]
@GregSutcliffe thanks will look at it.

**Comment by isratrade** [comment]
closed. replaced by pr #488

**Pull request closed without merging**